### ISSUE TITLE
Change the current algorithm to select problem in the suggested practice section 

### ISSUE THREAD

**Issue by shrinish123** [shrinish123/lets-Upsolve#8]
Right now the current algorithm just selects 100 points higher rated problems than your current rating. We are in the need to implement a better algorithm, Suggestion from your side are welcome .
One suggestion is to fetch problems that helped people to improve their rating and then as per user rating, suggest them to the user.

**Comment by MaYaNkKashyap681** [comment]
can we add something that predicts the problems according to the weak topics of any user?

**Comment by shrinish123** [comment]
Yes maybe you can do it. But what algorithm and API endpoints will you be using in order to predict the weaker topics of user ? 
You can work on it , if you are interested

**Comment by MaYaNkKashyap681** [comment]
We are making a graph for the user, so by using that graph we can find out the section he has solved the least.

1). In which file do you have the logic for this algorithm?

**Comment by MaYaNkKashyap681** [comment]
> 1). In which file do you have the logic for this algorithm?
In which file do i get this algorithm.

**Comment by shrinish123** [comment]
The graph is just from the data being fetched over the submissions API endpoint of the user. You can check the algorithm to make the graph in the Analysis Page component. 
But there is an issue in your algorithm, where you find a person's weak topic by on of the questions he has solved. For example , I have solved many  problems of DP because I am the weakest at it , but as per your algo it  will show I am strong at it. which contradicts the fact. So we need to make some better way to look at it .
Maybe a probable algo can be to check no of wrong submissions on every topic and where a particular person's accuracy is low, i.e.  correctSubmssionCount / Total SubmissionCount per topic , to predict the topic strength for the user. 
I guess this can  be a thing to implement in the analysis Page.Maybe will make  a issue on this topic. You can work on it if you would want to

**Comment by MaYaNkKashyap681** [comment]
Yes I want to work on this.
